### PR TITLE
Make pinv work for Adjoint

### LINKED ISSUE #578: Pinv of lazy adjoints and transposes?
The two methods for pinv of lazy transposes/adjoints in [stdlib/LinearAlgebra/src/adjtrans.jl](https://github.com/JuliaLang/julia/blob/5f2759d85eeeb9c68e35e5e84b10715a49f4449d/stdlib/LinearAlgebra/src/adjtrans.jl#L223) only apply to the transpose or adjoint of a vector, but not matrices. 

The Moore-Penrose pseudoinverse commutes with both conjugation, transposes, and conjugate transposes (adjoints). A naive implementation of pinv for Adjoints would be pinv(m::Adjoint) = pinv(m')' , and similarly with transpose for lazy transposes.

### LINKED ISSUE #29723: Pinv of lazy adjoints and transposes?
The two methods for pinv of lazy transposes/adjoints in [stdlib/LinearAlgebra/src/adjtrans.jl](https://github.com/JuliaLang/julia/blob/5f2759d85eeeb9c68e35e5e84b10715a49f4449d/stdlib/LinearAlgebra/src/adjtrans.jl#L223) only apply to the transpose or adjoint of a vector, but not matrices. 

The Moore-Penrose pseudoinverse commutes with both conjugation, transposes, and conjugate transposes (adjoints). A naive implementation of pinv for Adjoints would be pinv(m::Adjoint) = pinv(m')' , and similarly with transpose for lazy transposes.

### PR DESCRIPTION
**Pull request by andreasnoack**
Fixes JuliaLang/LinearAlgebra.jl#578

### CODE REVIEW

File: stdlib/LinearAlgebra/src/dense.jl [hunk 1]
```diff
@@ -1266,10 +1266,10 @@ function pinv(A::StridedMatrix{T}, rtol::Real) where T
     Sinv        = zeros(Stype, length(SVD.S))
     index       = SVD.S .> rtol*maximum(SVD.S)
     Sinv[index] = one(Stype) ./ SVD.S[index]
-    Sinv[findall(.!isfinite.(Sinv))] .= zero(Stype)
```
Comment (mbauman) [inline stdlib/LinearAlgebra/src/dense.jl]: Why is this change needed?

Comment (andreasnoack) [reply stdlib/LinearAlgebra/src/dense.jl]: It's not needed for this change. However, `zero(Stype)` isn't needed since the array is already allocated with `Stype` elements and `0` is easier to read than `zero(Stype)`.

Comment (mbauman) [reply stdlib/LinearAlgebra/src/dense.jl]: Aren't there places where `zero(T)` works but `convert(T, 0)` fails?  I'd let sleeping dogs lie here, especially since we also use `one`  and `zeros` immediately surrounding this line.

Comment (andreasnoack) [reply stdlib/LinearAlgebra/src/dense.jl]: There are but they don't apply here since `Stype` is the type the singular values. It's mainly for non-number types like arrays that you can't rely on converting from `0`. This change is unrelated to the PR so I've removed it.

**Review by mbauman — APPROVED** [review]

File: stdlib/LinearAlgebra/src/dense.jl (hunk 1 above)
Comment (andreasnoack) [reply stdlib/LinearAlgebra/src/dense.jl]: That is true. The singular values have units so that would fail. Unfortunately, it would be a bit complicated to make the svd solver in `GenericLinearAlgebra` work for `Unitful.Quantity` because it's only a subtype of `Number`.
